src/custom/pages/Claim/ClaimsTable.tsx

```diff
@@ -122,9 +124,18 @@ export default function ClaimsTable({
 }: ClaimsTableProps) {
   const { selectedAll, selected, activeClaimAccount, claimStatus, isInvestFlowActive } = useClaimState()
   const pendingClaimsSet = useAllClaimingTransactionIndices()
+  const { setSelectedAll } = useClaimDispatchers()
 
   const { deployment: start, investmentDeadline, airdropDeadline } = useClaimTimeInfo()
 
+  const paidClaims = getPaidClaims(userClaimData)
+
+  useEffect(() => {
+    if (selected.length === paidClaims.length) {
+      setSelectedAll(true)
+    }
```

Comment:
Shouldn't it be updated when it's not fully selected as well?

```suggestion
  setSelectedAll(selected.length === paidClaims.length)
```

Reply:
No because if you unselect one option it will set selectedAll to false.

Reply:
...which is what we want 😂

Reply:
Yes